gateway/handle.go

```diff
@@ -684,72 +696,70 @@ func (gw *Handle) internalBatchHandlerFunc() http.HandlerFunc {
 		gw.stats.NewTaggedStat(
 			"gateway.write_key_failed_requests",
 			stats.CountType,
-			gw.newSourceStatTagsWithReason(arctx, reqType, errorMessage),
+			gw.newReqTypeStatsTagsWithReason(reqType, errorMessage),
 		).Increment()
 		gw.logger.Infon("response",
 			logger.NewStringField("ip", kithttputil.GetRequestIP(r)),
 			logger.NewStringField("path", r.URL.Path),
 			logger.NewIntField("status", int64(status)),
 			logger.NewStringField("body", responseBody),
 		)
+		gw.logger.Debugn("response",
+			logger.NewStringField("ip", kithttputil.GetRequestIP(r)),
+			logger.NewStringField("path", r.URL.Path),
+			logger.NewIntField("status", int64(status)),
+			logger.NewStringField("body", responseBody),
+			logger.NewStringField("request", string(body)),
+		)
 		http.Error(w, responseBody, status)
 	}
 }
 
-func (gw *Handle) extractJobsFromInternalBatchPayload(
-	arctx *gwtypes.AuthRequestContext,
-	reqType string,
-	body []byte,
-) ([]*jobsdb.JobT, error) {
-	if !gjson.ValidBytes(body) {
-		return nil, fmt.Errorf("%s", response.InvalidJSON)
+func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byte) ([]*jobsdb.JobT, error) {
+	type params struct {
+		MessageID       string `json:"message_id"`
+		SourceID        string `json:"source_id"`
+		SourceJobRunID  string `json:"source_job_run_id"`
+		SourceTaskRunID string `json:"source_task_run_id"`
+		UserID          string `json:"user_id"`
+		TraceParent     string `json:"traceparent"`
 	}
-	gw.requestSizeStat.Observe(float64(len(body)))
 
-	type jobObject struct {
-		userID     string
-		events     []map[string]interface{}
-		receivedAt string
+	type singularEventBatch struct {
```

Comment:
Not related to the changes. But this is being created twice. https://github.com/rudderlabs/rudder-server/blob/c2acf9f91b307b0e78ab4eed80afbb5e01c84c18/gateway/handle.go#L470

Can we create it outside?

Reply:
Good point, This might require some refactoring, we can pick this up separately.